subworkflows/local/input_check.nf

```diff
@@ -11,25 +11,27 @@ workflow INPUT_CHECK {
     main:
     SAMPLESHEET_CHECK ( samplesheet )
         .csv
-        .splitCsv ( header:true, sep:',' )
+        .splitCsv ( header:true )
         .map { get_samplesheet_paths(it) }
-        .set { reads }
+        .set { meta }
 
-    emit:
-    reads                                     // channel: [ val(meta), [ reads ] ]
+    emit: meta                  // channel: [ val(metadata), [ files ] ]
     versions = SAMPLESHEET_CHECK.out.versions // channel: [ versions.yml ]
 }
 
 // Function to get list of [ meta, filenames ]
 def get_samplesheet_paths(LinkedHashMap row) {
+    //---------
+    // Save sample, alleles, mhc_class and file_type in a dictionary (metadata)
+    // and return a list of meta and the filename.
+    //---------
 
     def allele_string = generate_allele_string(row.alleles, row.mhc_class)
```

Comment:
What we actually do here as well is to construct the allele string in case an allele file is provided. I did not see this in the `check_samplesheet` (where it also does not necessarily belong).